src/main/java/org/apache/mesos/state/SchedulerState.java

```diff
@@ -0,0 +1,65 @@
+package org.apache.mesos.state;
+
+import org.apache.mesos.Protos.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.*;
+
+/**
+ * Read/write interface for the state of a Scheduler.
+ */
+public class SchedulerState  {
+    private static final Logger log = LoggerFactory.getLogger(SchedulerState.class);
+
+    private static final String SUPPRESSED_KEY = "suppressed";
+    private final StateStore stateStore;
+
+    public SchedulerState(StateStore stateStore) {
+        this.stateStore = stateStore;
+    }
+
+    public Optional<FrameworkID> getFrameworkId() {
```

Comment:
This and setFrameworkId() feel unrelated to the rest of the class. Could the caller just invoke the stateStore functions directly?